### DISCUSSION TITLE
About country fields

### DISCUSSION THREAD

**Discussion by Philmo67** [benjaminjonard/koillection#1012] [Q&A]
Hello,
I'm setting up Koillection to track my wine cellar and spirits so each item should have a country field.
Inconveniences I encouter using this field type : 
- when selecting the country, it doesn't seem to be possible to type a country letter or name, you have to scroll down through the dropbox
- it doesn't seem to be possible to search through the countries of the collections items even if I enabled data search
Am I missing something ? Could this be improved ?

**Comment by benjaminjonard** [comment]
Hello, you should be able to search countries by name

![Screenshot from 2023-12-15 09-50-17](https://github.com/benjaminjonard/koillection/assets/20560781/126fde3a-acc1-46d0-8bf2-9716df7e264a)

Maybe I could make it a bit easier by focusing the search input when opening the select input

For the item search, you are right it doesn't apply to country fields. Only the country code is stored in the database, not the name (it makes it easier to properly display country names based on the user language). 
I see the use case, I need to think about how to best implement this. Maybe a country filter on the search result page would be better than plain text search ?

**Comment by Philmo67** [comment]
Thank you, indeed it was a focus "problem", I didn't notice there was a search field.
Search would have to be made on the translated country label, so maybe a filter would be a good solution.
It seems this solution could also apply to boolean fields.
